auth0-lock.ts

```diff
@@ -53,9 +61,22 @@ export class AuthService {
   }
 }
 
-export function provideAuthService(clientId: string, domain: string): Provider {
-  return {
-    provide: AuthService,
-    useValue: new AuthService(clientId, domain)
-  };
+export function provideAuthService(clientId: string, domain: string): Provider;
+export function provideAuthService(clientId: string, domain: string, homePath: string, redirectPath: string): Provider;
+export function provideAuthService(clientId: string, domain: string, homePath?: string, redirectPath?: string): Provider {
+  if(redirectPath == null){

```

Comment:
Should this check be on `homePath`? That is the one that requires `router` dependency.